### DISCUSSION TITLE
Update/Transition Vue 3, TypeScript & Vite

### DISCUSSION THREAD

**Discussion by pfmephisto** [linkarkitektur/SpeckleLCA#15] [General]
I propose upgrading/transitioning the project to Vue 3 using Typescript and the Vite as webserve for local development.
The current vue 2 version is long outdate and type safety, will make things clearer, especially with multiple people working on the project.

**Comment by pfmephisto** [comment]
Another reason for updating would also be all the security warning for all the outdate packages that we are using.

**Reply by pfmephisto** [reply]
Also consider [nuxt](https://nuxt.com/) and [tRPC](https://trpc.io/)

**Comment by AldenChaseyor3** [comment]
thanks for sharing!

**Comment by pfmephisto** [comment]
There is a branch now. It's not quite up to date, but it's also not functional yet.
So I am going to continue porting the old conde and then pulling in the changes when I get to them.
https://github.com/linkarkitektur/SpeckleLCA/tree/vue3-vite-typescript-transition

**Comment by vwnd** [comment]
Hi @pfmephisto are you still working on this? Happy to help 😊

**Reply by pfmephisto** [reply]
Hi @vwnd ,
@yschindel from H&D is currently looking in to it, but I'm sure he would not say no to some help.

**Reply by vwnd** [reply]
Cool! @yschindel let me know if you need anything :)

**Reply by yschindel** [reply]
@pfmephisto @vwnd I had to stop the refactoring, we are doing some testing on how to handle buildups and calculation results. Since I did not get very far yet, there is not much to loose. So feel free to start with the "clean_start" branch yourself. 
We can talk about how we can integrate our tests in one of the future meetings.

**Reply by vwnd** [reply]
Alright. I'm gonna give it a try on this!

@pfmephisto I notice the UI currently uses Vuetify which is in beta support for Vue 3. Would you rather stick with Vuetify in Beta or switch to something else that is compatible with Vue 3 (e.g. Tailwindcss)?

**Reply by pfmephisto** [reply]
@vwnd in the long run my plan was to change it maybe to Tailwind. But if I would have done the work myself I would have used vuetify so as not to make rewriting the page more complex for me. But I you are taking a stab at it and want to do Tailwind right away, be my guest :smiley:

**Reply by vwnd** [reply]
Gonna try to stick with Vuetify, the update in itself is already a lot of work 😄
